templates/deployment.yaml

```diff
@@ -133,12 +138,14 @@ spec:
         {{- toYaml . | nindent 8 }}
     {{- end }}
     {{- if .Values.topologySpreadConstraints.enabled }}
-      topologySpreadConstraints: 
-        - labelSelector:
+      topologySpreadConstraints:
+        {{- range .Values.topologySpreadConstraints.topologyKeys }}
+        - topologyKey: {{ . | quote }}
+          labelSelector:
             matchLabels:
-              app: {{ .Release.Name }}
-              type: {{ .Chart.Name }}
-          maxSkew: {{ .Values.topologySpreadConstraints.maxSkew }}
-          topologyKey: {{ .Values.topologySpreadConstraints.topologyKey | quote }}
-          whenUnsatisfiable: {{ .Values.topologySpreadConstraints.whenUnsatisfiable }}
+              app: {{ $.Release.Name }}
```

Comment:
no idea what the '$' is for :-D, but it's probably ok :-)

Reply:
@rbapst-tamedia It's to reference the "parent" context when using `range` --- [helm docs](https://helm.sh/docs/chart_template_guide/control_structures/)